src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/order_actions.html.twig

```diff
@@ -62,11 +64,79 @@
   {% if orderForViewing.documents.canGenerateDeliverySlip %}
     <form class="form-inline d-inline-block form-horizontal ml-2">
       <div class="input-group">
-        <a href="{{ path('admin_orders_generate_delivery_slip_pdf', {'orderId': orderForViewing.id}) }}" class="btn btn-action">
+        <a href="{{ path('admin_orders_generate_delivery_slip_pdf', {'orderId': orderForViewing.id}) }}"
+           class="btn btn-action">
           <i class="material-icons">local_shipping</i>
           {{ 'View delivery slip'|trans({}, 'Admin.Orderscustomers.Feature') }}
         </a>
       </div>
     </form>
   {% endif %}
+
+  {# Rendering of hook actionGetBackOfficeOrderButtons, see ActionsBarButtonsCollection #}
+  {% if (backOfficeOrderButtons.isEmpty() != true) %}
+
+    {% set buttons = backOfficeOrderButtons.getButtons() %}
+    {% if buttons|length > 2 %}
+
+      {% set firstButton = buttons|first %}
+      <div class="d-inline-block ml-2">
+        <div class="input-group">
+          <a class="btn {{ firstButton.class }}"
+          {% for tagName, tagContent in firstButton.tags %}
+            {{ tagName }}="{{ tagContent }}"
+          {% endfor %}
+          >
+          {{ firstButton.content|raw }}
+          </a>
+        </div>
+      </div>
+
+      {# Display other buttons into a dropdown #}
+      {% set buttons = buttons|slice(1, buttons|length) %}
+
+      <div class="d-inline-block ml-2">
+        <div class="input-group">
+          <div class="dropdown">
+            <button class="btn btn-action dropdown-toggle" type="button" id="dropdownOrderActionBar"
+                    data-toggle="dropdown"
+                    aria-haspopup="true" aria-expanded="false">
+              <i class="material-icons form-error-icon">more_horiz</i>
+
+              {{ 'More actions'|trans({}, 'Admin.Global') }}
+            </button>
+
+            <div class="dropdown-menu" aria-labelledby="dropdownOrderActionBar">
+              {% for button in buttons %}
+                <a class="dropdown-item btn {{ button.class }}"
```

Comment:
Since the only change between these buttons and when there are only two is the extra class `dropdown-item` maybe you can create a twig template dedicated to render this button You can manage the extra class either through a parameter of the template Or by adding it in twig right before rendering the template `set button.class = button.class ~ ' dropdown-item'`

This would have few advantages, factorize this code, make the `order_actions.html.twig` template shorter and finally allow to override the button template if needed

Reply:
If I remember correctly, `include` statements in Twig are not the best optimized :/ so it would be better to use a macro here. Do you think it's worth it ? I've extracted all the "hook extra buttons" into a dedicated Twig template, is it enough ?